pkg/common/scheduler/statefulset/state_test.go

```diff
@@ -76,6 +78,62 @@ func TestStateBuilder(t *testing.T) {
 			freec:           int32(24),
 			schedulerPolicy: MAXFILLUP,
 		},
+		{
+			name: "many vpods, with gaps and reserved vreplicas",
+			vpods: [][]duckv1alpha1.Placement{
+				{{PodName: "statefulset-name-0", VReplicas: 1}, {PodName: "statefulset-name-2", VReplicas: 5}},
+				{{PodName: "statefulset-name-1", VReplicas: 0}},
+				{{PodName: "statefulset-name-1", VReplicas: 0}, {PodName: "statefulset-name-3", VReplicas: 0}},
+			},
+			expected: state{capacity: 10, free: []int32{int32(4), int32(10), int32(5), int32(10)}, lastOrdinal: 2, schedulerPolicy: MAXFILLUP},
+			freec:    int32(19),
+			reserved: map[types.NamespacedName]map[string]int32{
+				{Name: "s1", Namespace: testNs}: {
```

Comment:
Can we also changes this to the other format vpod-name-# to avoid future confusion maybe?